Application/EdFi.Ods.AdminApi/Features/OdsInstanceDerivative/EditOdsInstanceDerivative.cs

```diff
@@ -0,0 +1,109 @@
+    {
+        request.Id = id;
+        SetCurrentConnectionString(db, request, id);
+        await validator.GuardAsync(request);
+        editOdsInstanceDerivativeCommand.Execute(request);
+        return Results.Ok();
+    }
+
+    private void SetCurrentConnectionString(IUsersContext db, EditOdsInstanceDerivativeRequest request, int id)
+    {
+        if (string.IsNullOrEmpty(request.ConnectionString))
+            request.ConnectionString = db.OdsInstanceDerivatives.Find(id)?.ConnectionString;
+    }
+
+    [SwaggerSchema(Title = "EditOdsInstanceDerivativeRequest")]
+    public class EditOdsInstanceDerivativeRequest : IEditOdsInstanceDerivativeModel
+    {
+        [SwaggerSchema(Description = FeatureConstants.OdsInstanceDerivativeIdDescription, Nullable = false)]
+        public int Id { get; set; }
+        [SwaggerSchema(Description = FeatureConstants.OdsInstanceDerivativeOdsInstanceIdDescription, Nullable = false)]
+        public int OdsInstanceId { get; set; }
+        [SwaggerSchema(Description = FeatureConstants.OdsInstanceDerivativeDerivativeTypeDescription, Nullable = false)]
+        public string? DerivativeType { get; set; }
+        [SwaggerSchema(Description = FeatureConstants.OdsInstanceDerivativeConnectionStringDescription, Nullable = false)]
+        public string? ConnectionString { get; set; }
+    }
+
+    public class Validator : AbstractValidator<EditOdsInstanceDerivativeRequest>
+    {
+        private readonly IGetOdsInstanceQuery _getOdsInstanceQuery;
+        private readonly string _databaseEngine;
+        public Validator(IGetOdsInstanceQuery getOdsInstanceQuery, IOptions<AppSettings> options)
+        {
+            _getOdsInstanceQuery = getOdsInstanceQuery;
+            _databaseEngine = options.Value.DatabaseEngine ?? throw new NotFoundException<string>("AppSettings", "DatabaseEngine");
+
+            RuleFor(m => m.DerivativeType).NotEmpty();
+
+            RuleFor(m => m.DerivativeType)
+                .Matches("^(ReadReplica|Snapshot)$")
+                .WithMessage(FeatureConstants.OdsInstanceDerivativeDerivativeTypeNotValid)
+                .When(m => !string.IsNullOrEmpty(m.DerivativeType));
+
+            RuleFor(m => m.OdsInstanceId)
+                .NotEqual(0)
+                .WithMessage(FeatureConstants.OdsInstanceIdValidationMessage);
+
+            RuleFor(m => m.OdsInstanceId)
+                .Must(BeAnExistingOdsInstance)
+                .When(m => !m.OdsInstanceId.Equals(0));
+
+            RuleFor(m => m.ConnectionString)
+                .Must(BeAValidConnectionString)
+                .WithMessage(FeatureConstants.OdsInstanceConnectionStringInvalid)
+                .When(m => !string.IsNullOrWhiteSpace(m.ConnectionString));
+        }
+
+        private bool BeAnExistingOdsInstance(int id)
+        {
+            try
```

Comment:
@lchavarriaga
This "try-catch" code block is no longer necessary since the  "_getOdsInstanceQuery.Execute(id)" is throwing the exception when the odsInstance is not found